vendor/bat-native-ledger/src/bat/ledger/internal/contribution/contribution_unblinded.cc

```diff
@@ -131,176 +114,405 @@ std::string GenerateTokenPayload(
   return json;
 }
 
-void Unblinded::Start(const std::string& viewing_id) {
+bool GetStatisticalVotingWinner(
+    double dart,
+    const double amount,
+    std::vector<ledger::ContributionPublisher> list,
+    braveledger_contribution::Winners* winners) {
+  if (!winners) {
+    return false;
+  }
+
+  double upper = 0.0;
+  for (auto& item : list) {
+    upper += item.total_amount / amount;
+    if (upper < dart) {
+      continue;
+    }
+
+    auto iter = winners->find(item.publisher_key);
+
+    uint32_t current_value = 0;
+    if (iter != winners->end()) {
+      current_value = winners->at(item.publisher_key);
+      winners->at(item.publisher_key) = current_value + 1;
+    } else {
+      winners->emplace(item.publisher_key, 1);
+    }
+
+    return true;
+  }
+
+  return false;
+}
+
+void GetStatisticalVotingWinners(
+    uint32_t total_votes,
+    const double amount,
+    ledger::ContributionPublisherList list,
```

Comment:
and here

Reply:
this is list of pointers, so not needed

Reply:
without passing it by reference(which is only one processor signal) you create a new vector of elements here with pointers(each of it uses a one processor signal as well) and everything consumes additional memory which is not needed at all. Why do you need to copy that vector?

Reply:
done